components/Layout.tsx

```diff
@@ -61,41 +64,96 @@ export default function Layout({
   children: React.ReactNode;
 }) {
   const [mobileOpen, setMobileOpen] = useState(false);
+  const [searchIsOpen, setSearchOpen] = useState(false);
+  const [searchTerm, setSearchTerm] = useState('');
 
   const { width } = useWindowSize();
-  const { asPath } = useRouter();
+  const router = useRouter();
+  const asPath = router.asPath;
 
   const handleDrawerToggle = () => {
     setMobileOpen(!mobileOpen);
   };
 
   const drawer = (
-    <List>
-      {[...navItems]
-        .sort((a, b) => {
-          if (a.title === 'Home') {
-            return -1;
-          }
-          if (b.title === 'Home') {
-            return 1;
-          }
-          return a.title.toLowerCase() > b.title.toLowerCase() ? 1 : -1;
-        })
-        .map(({ title: itemTitle, href, Icon }) => (
+    <List id='nav-sidebar'>
+      {[
+        {
+          title: 'Home',
+          href: '/',
+          Icon: HomeIcon,
+        },
+        {
+          title: 'Search',
+          href: '',
+          Icon: SearchIcon,
+        },
+      ].map(({ title: itemTitle, href, Icon }) => (
+        <>
           <ListItem
-            key={href}
             disablePadding
+            key={itemTitle.toLowerCase()}
           >
             <ListItemButton
-              href={href}
-              component={Link}
-              disabled={asPath === href}
+              {...(itemTitle === 'Search'
+                ? { onClick: () => setSearchOpen(!searchIsOpen) }
+                : {
+                    href: href,
+                    component: Link,
+                  })}
             >
               <ListItemIcon>{!!Icon && <Icon />}</ListItemIcon>
               <ListItemText primary={itemTitle} />
             </ListItemButton>
           </ListItem>
-        ))}
+          {itemTitle === 'Search' && (
+            <ListItem
+              sx={{
+                display: searchIsOpen ? 'initial' : 'none',
+              }}
+            >
+              <TextField
+                variant='standard'
+                placeholder='Search...'
+                onChange={(e) => setSearchTerm(e.target.value)}
+                onKeyDown={(e) => {
+                  const curHref = navItems.filter(
+                    ({ title: _itemTitle }) => {
+                      return _itemTitle
+                        .toLowerCase()
+                        .includes(searchTerm.toLowerCase());
+                    },
+                  )[0]?.href;
+                  if (e.keyCode === 13 && curHref !== null) {
+                    router.push(String(curHref));
+                  }
+                  if ([38, 40].indexOf(e.keyCode) > -1) {
+                    e.preventDefault();
+                  }
+                }}
+              />
+            </ListItem>
+          )}
+        </>
+      ))}
+      {[...navItems].sort().map(({ title: itemTitle, href, Icon }) => {
```

Comment:
The `sort` here does nothing because it's sorting an array of objects, this may have been the way it was when you forked, but it has been updated since.

It may be easier to just grab the latest `Layout.tsx` from `dev` and add the search field onto that?

Reply:
Sure thing, I'll merge upstream